conans/client/export.py

```diff
@@ -69,16 +67,20 @@ def _init_export_folder(destination_folder):
     return previous_digest
 
 
-def _export(file_patterns, origin_folder, destination_folder, output):
-    file_patterns = file_patterns or []
+def _export(conanfile, origin_folder, destination_folder, output):
+    exports = conanfile.exports or []
+    exports_sources = conanfile.exports_sources or []
     try:
         os.unlink(os.path.join(origin_folder, CONANFILE + 'c'))
     except:
         pass
 
     copier = FileCopier(origin_folder, destination_folder)
-    for pattern in file_patterns:
-        copier(pattern)
+    copied_files = []
+    for pattern in exports:
+        partial_copied_files = copier(pattern)
+        if pattern in exports_sources:
```

Comment:
If the pattern is in export_sources, I annotate that and store it in the "conan_sources.txt" file to know which files put in each tgz.